### PR TITLE
Feature/client singleton methods

### PR DESCRIPTION
**Pull request by lkatzman**
- Add `singleton` methods to make add endpoints to client
- Update tests to include `singleton` changes

### CODE REVIEW

**Review by erikgrueter1 — APPROVED** [review]
@lkatzman Run linters, test this branch in our application and ensure all tests are passing. Then merge and draft tag

**Comment by bernEsp** [comment]
Thanks @erikgrueter1 @lkatzman I look forward to see these changes on master and the new gem version.

File: lib/yelp/fusion/singleton.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+module Yelp
+  module Fusion
+    module Singleton
+      # These methods were copied from the old yelp implementation
```
Comment (erikgrueter1) [inline lib/yelp/fusion/singleton.rb]: @lkatzman Remove this. These methods rely on completely new code. We don't need to indicate every place our gem bears resemblance to the old gem.

File: lib/yelp/fusion/singleton.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      REQUEST_CLASSES = [Yelp::Fusion::Endpoint::Search,
+                         Yelp::Fusion::Endpoint::Business,
+                         Yelp::Fusion::Endpoint::Phone,
+                         Yelp::Fusion::Endpoint::Review,
+                         Yelp::Fusion::Endpoint::Transaction,
+                         Yelp::Fusion::Endpoint::Match].freeze
+
+      private
+
+      # This goes through each endpoint class and creates singletone methods
```
Comment (erikgrueter1) [inline lib/yelp/fusion/singleton.rb]: singletone is should be spelled singleton

**Review by erikgrueter1 — CHANGES_REQUESTED** [review]

File: lib/yelp/fusion/singleton.rb (hunk 2 above)
Comment (bernEsp) [reply lib/yelp/fusion/singleton.rb]: +1

**Review by erikgrueter1 — APPROVED** [review]
@lkatzman  Run linters, ensure tests are passing, then tag to v0.3-beta and push to Ruby gems. Notify @bernardogalindo When this is done so he can update his project.

**Pull request merged**